src/main/scala/vaquita/VecALU.scala

```diff
@@ -0,0 +1,462 @@
+          val comparison = comparison_operators(vs1_elem.asSInt, vs2_elem.asSInt)
+          comp_1b(counter) := (io.mask_arith && comparison) || (!io.mask_arith && comparison && vs0_mask(counter))
+          comp_vs3(counter) := (!vs0_mask(counter) && !io.mask_arith)
+          comp_0b(counter) := (io.mask_arith && !comparison)
+          comparison_vec_bit_wires(counter) := MuxCase(0.B, Array(
+            (comp_0b(counter) === 1.B) -> 0.B,
+            (comp_vs3(counter) === 1.B) -> vs3_bit(counter),
+            (comp_1b(counter) === 1.B) -> 1.B
+          ))
+          counter += 1
+        }
+      }
+    }
+    comparison_vec_bit_wires.asUInt
+  }
+
+  def Arithmatic(vs1_in: SInt, vs2_in: SInt,vs3:SInt,sew:Int,v0_bit_mask:UInt): SInt = {
+    val maxUInt32 = "hFFFFFFFF".U(32.W)
+    val sum = WireInit(0.S(33.W))
+    val sub = WireInit(0.S(33.W))
+    val maxValue = (1.S << (sew - 1)) - 1.S 
+    val minValue = -(1.S << (sew - 1)) 
+    sub := (vs2_in -& vs1_in)
+    sum := (vs1_in.asUInt +& vs2_in.asUInt).asSInt
+    // Overflow for addition
+    val positiveOverflowAdd = (vs1_in(sew - 1) === 0.U && vs2_in(sew - 1) === 0.U && sum(sew - 1) === 1.U)
+    val negativeOverflowAdd = (vs1_in(sew - 1) === 1.U && vs2_in(sew - 1) === 1.U && sum(sew - 1) === 0.U)
+    // Overflow for subtraction
+    val positiveOverflowSub = (vs2_in(sew - 1) === 0.U && vs1_in(sew - 1) === 1.U && sub(sew - 1) === 1.U)
+    val negativeOverflowSub = (vs2_in(sew - 1) === 1.U && vs1_in(sew - 1) === 0.U && sub(sew - 1) === 0.U)
+    val lookuptable = Seq(
+      0.U         -> (vs1_in + vs2_in),//add
+      2.U         -> (vs2_in - vs1_in),//sub
+      3.U         -> (vs1_in - vs2_in),//rsub
+      9.U         -> (vs1_in & vs2_in),// and
+      10.U        -> (vs1_in | vs2_in),//or
+      11.U        -> (vs1_in ^ vs2_in),//xor
+      37.U        -> (vs2_in << (vs1_in.asUInt%sew.U)), //vsll
+      "b101000".U -> (vs2_in.asUInt >> (vs1_in.asUInt%sew.U)).asSInt, //vsrl
+      "b101001".U -> ((vs2_in >> (vs1_in.asUInt%sew.U)).asSInt), //vsra
+      "b010111".U -> (vs1_in), //vmv 
+      "b000100".U -> Mux(vs1_in.asUInt < vs2_in.asUInt,vs1_in.asUInt,vs2_in.asUInt).asSInt,//minu
+      "b000101".U -> Mux(vs1_in < vs2_in,vs1_in,vs2_in),//min
+      "b000110".U -> Mux(vs1_in.asUInt > vs2_in.asUInt,vs1_in.asUInt,vs2_in.asUInt).asSInt,//maxu
+      "b000111".U -> Mux(vs1_in > vs2_in,vs1_in,vs2_in),//max
+      "b100000".U -> Mux(sum(32), "hFFFFFFFF".U, sum(31,0)).asSInt,//vsaddu
+      "b100001".U -> (Mux(positiveOverflowAdd, maxValue, Mux(negativeOverflowAdd, minValue, sum))),//vsadd
+      "b100010".U -> Mux(vs2_in.asUInt < vs1_in.asUInt, 0.U,vs2_in.asUInt - vs1_in.asUInt ).asSInt,//vssubu
+      "b100011".U -> Mux(positiveOverflowSub, maxValue, Mux(negativeOverflowSub, minValue, sub(31, 0).asSInt)),//vssub
+      "b010000".U -> (vs1_in.asUInt + vs2_in.asUInt + v0_bit_mask).asSInt, //vadc
+      "b010010".U -> (vs2_in.asUInt - vs1_in.asUInt - v0_bit_mask).asSInt //vsbc
```

Comment:
I think a better approach would be to follow the Rocket Chip logic for saving all the func bits for each operations separately in a Singleton Object and then use its members here. will bring a lot more readability to the code